exporter/collector/metricsexporter.go

```diff
@@ -19,20 +19,29 @@ package collector
 
 import (
 	"context"
+	"fmt"
 
 	monitoring "cloud.google.com/go/monitoring/apiv3/v2"
 	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/exporter/exporterhelper"
 	"go.opentelemetry.io/collector/model/pdata"
-	"google.golang.org/genproto/googleapis/api/metric"
+	metricpb "google.golang.org/genproto/googleapis/api/metric"
 	monitoredrespb "google.golang.org/genproto/googleapis/api/monitoredres"
 	monitoringpb "google.golang.org/genproto/googleapis/monitoring/v3"
+	"google.golang.org/protobuf/types/known/timestamppb"
 )
 
 // metricsExporter is the GCM exporter that uses pdata directly
 type metricsExporter struct {
 	cfg    *Config
 	client *monitoring.MetricClient
+	mapper metricMapper
+}
+
+// metricMapper is the part that transforms metrics. Separate from metricsExporter since it has
+// all pure functions.
+type metricMapper struct {
+	cfg *Config
```

Comment:
cfg doesn't appear to be used anywhere.

Reply:
You're right, i added it preemptively. It will effect some of the mapping logic like metric name. I can remove it from this PR until it is needed if you prefer

Reply:
no, thats fine.